core/server/models/settings.js

```diff
@@ -163,17 +166,31 @@ Settings = ghostBookshelf.Model.extend({
 
         return this.findAll(options).then(function then(allSettings) {
             var usedKeys = allSettings.models.map(function mapper(setting) { return setting.get('key'); }),
-                insertOperations = [];
+                insertOperations = [],
+                isFirstRun = false;
 
             _.each(getDefaultSettings(), function each(defaultSetting, defaultSettingKey) {
                 var isMissingFromDB = usedKeys.indexOf(defaultSettingKey) === -1;
+
                 if (isMissingFromDB) {
-                    defaultSetting.value = defaultSetting.defaultValue;
+                    debug('isMissingFromDB', defaultSettingKey);
+                    // @TODO revisit this? maybe an even better way to do this with model events etc?
+                    if (defaultSettingKey === 'dbHash') {
```

Comment:
We could move this up to the `parseDefaultSettings` function, which makes it more readable i think.
We can define a default fallback object.

```
var fallback = {
   'dbHash': uuid.v4()
}
_.each(settings, function each(setting, settingName) {
  setting.value = setting.defaultValue || fallback(setting.key);
});
```